### PR TITLE
UIAPPS-167 Add dashboard cog menu and side panel

### PR DESCRIPTION
**Pull request by joneshf-dd**
Motivation
----------

- Jira Issue: https://datadoghq.atlassian.net/browse/UIAPPS-167
- We want the kitchen sink to showcase everything. We weren't showing either the dashboard cog menu or the side panel feature.

Changes
-------

- We add the missing features for the kitchen sink.

Testing
-------

- This is likely best tested once we have support for templates in the Datadog UI. We are almost assuredly going to need to make adjustments to this template afteward.

Releases
--------

Choose one:

-   [x] No release is necessary.
        If you're only updating examples/documentation, this is likely what you want.
-   [ ] All packages that need a release have a changeset.

### CODE REVIEW

**Comment by joneshf-dd** [comment]
Current dependencies on/for this PR:
* master
  * **PR #142** <a href="https://app.graphite.dev/github/pr/DataDog/apps/142" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DataDog/apps/142?utm_source=stack-comment).

File: examples/kitchen-sink/public/_redirects [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /custom-widget    /index.html   200
 /modal    /index.html   200
+/side-panel    /index.html   200
```
Comment (nmuldavin) [inline examples/kitchen-sink/public/_redirects:3]: This is unrelated to this change so consider it non-blocking, but don't we probably want some catch-all here `/*  /index.html   200` ?

**Review by nmuldavin — APPROVED** [review]
Looks great!

File: examples/kitchen-sink/public/_redirects (hunk 1 above)
Comment (joneshf-dd) [reply examples/kitchen-sink/public/_redirects:3]: I'm down to do whatever, but I honestly don't know. I've just been cargo-culting this stuff since day one. There was a related question about the purpose of the file the other day: https://github.com/DataDog/apps/pull/138#discussion_r846095391.

**Comment by joneshf-dd** [comment]
Thanks for the review!

Going to merge for now, but more than down to circle back about the redirect stuff.
